### PR TITLE
Add Eth-wallets support

### PR DESCRIPTION
**Pull request by Karkunow**
This PR adds an example of the Eth-wallets integration from the Near Wallet Selector and [NEP-518](https://github.com/near/NEPs/issues/518) support.
See the additional docs here:
https://github.com/near/wallet-selector/blob/9a9f9535ec57633b949872bb51903e7802beb3e8/packages/ethereum-wallets/README.md

### CODE REVIEW

File: frontend/src/wallets/web3modal.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public: { http: [EVMWalletChain.ethRpcForNear] },
+    },
+    blockExplorers: {
+    default: {
+        name: "NEAR Explorer",
+        url: EVMWalletChain.walletExplorerUrl,
+    },
+    },
+    testnet: !onMainnet,
+};
```
Comment (frol) [inline frontend/src/wallets/web3modal.js]: Please, fix the formatting here

File: frontend/src/wallets/web3modal.js
Comment (frol) [inline frontend/src/wallets/web3modal.js:1]: Please, format this file with 2-spaces indentation

**Review by frol — COMMENTED** [review]
This looks good to me overall. Sorry for bothering you with the formatting nits.

@gagdiez Please, review as well

**Comment by Karkunow** [comment]
@frol done with the formatting fixes

File: frontend/src/wallets/near.js [hunk 2]
```diff
@@ -17,13 +20,11 @@ export class Wallet {
    * @constructor
    * @param {Object} options - the options for the wallet
    * @param {string} options.networkId - the network id to connect to
-   * @param {string} options.createAccessKeyFor - the contract to create an access key for
    * @example
-   * const wallet = new Wallet({ networkId: 'testnet', createAccessKeyFor: 'contractId' });
+   * const wallet = new Wallet({ networkId: 'testnet' });
```
Comment (gagdiez) [inline frontend/src/wallets/near.js]:
```suggestion
   * const wallet = new Wallet({ networkId: 'testnet', createAccessKeyFor: 'contractId' });
```

lets bring `createAccessKeyFor` back here, as it is a valid case (people might want to add a key for their specific project

File: frontend/src/wallets/near.js [hunk 3]
```diff
@@ -59,7 +65,7 @@ export class Wallet {
    * Displays a modal to login the user
    */
   signIn = async () => {
-    const modal = setupModal(await this.selector, { contractId: this.createAccessKeyFor });
+    const modal = setupModal(await this.selector, { contractId: "" });
```
Comment (gagdiez) [inline frontend/src/wallets/near.js]:
```suggestion
    const modal = setupModal(await this.selector, { contractId: this.createAccessKeyFor });
```

same here

File: frontend/src/wallets/near.js [hunk 4]
```diff
@@ -17,13 +20,11 @@ export class Wallet {
...
    * @param {string} options.networkId - the network id to connect to
-   * @param {string} options.createAccessKeyFor - the contract to create an access key for
    * @example
-   * const wallet = new Wallet({ networkId: 'testnet', createAccessKeyFor: 'contractId' });
+   * const wallet = new Wallet({ networkId: 'testnet' });
    * wallet.startUp((signedAccountId) => console.log(signedAccountId));
    */
-  constructor({ networkId = 'testnet', createAccessKeyFor = undefined }) {
-    this.createAccessKeyFor = createAccessKeyFor;
+  constructor({ networkId = 'testnet'}) {
```
Comment (gagdiez) [inline frontend/src/wallets/near.js]:
```suggestion
  constructor({ networkId = 'testnet'}, createAccessKeyFor = undefined }) {
    this.createAccessKeyFor = createAccessKeyFor;
```

File: frontend/src/wallets/near.js [hunk 5]
```diff
@@ -33,9 +34,14 @@ export class Wallet {
...
    */
   startUp = async (accountChangeHook) => {
+    const alwaysOnboardDuringSignIn = true;
     this.selector = setupWalletSelector({
       network: this.networkId,
-      modules: [setupMyNearWallet(), setupHereWallet()]
+      modules: [
+        setupMyNearWallet(),
+        setupHereWallet(),
+        setupEthereumWallets({ wagmiConfig, web3Modal, alwaysOnboardDuringSignIn }),
```
Comment (gagdiez) [inline frontend/src/wallets/near.js]:
```suggestion
        setupEthereumWallets({ wagmiConfig, web3Modal, alwaysOnboardDuringSignIn: true }),
```

maybe we just add the `true` inside of the object

**Review by gagdiez — CHANGES_REQUESTED** [review]

**Comment by gagdiez** [comment]
almost there! small changes request and we can merge this @Karkunow

**Review by gagdiez — APPROVED** [review]

**Pull request merged**